applets/notification_area/na-grid.c

```diff
@@ -275,6 +344,14 @@ na_grid_unrealize (GtkWidget *widget)
   GTK_WIDGET_CLASS (na_grid_parent_class)->unrealize (widget);
 }
 
+static void
+na_grid_size_allocate (GtkWidget     *widget,
+                       GtkAllocation *allocation)
+{
+  refresh_grid (NA_GRID (widget));
```

Comment:
Hmm, I would call `refresh_grid` after chaining-up. Otherwise `gtk_widget_get_allocation` might return previous allocation (too lazy to check if that is really true).

Reply:
I tried to queue it first, hello disco panel. ;) It works once in a while, but in many cases it causes heavy flicker on loading..

Reply:
queue?

Reply:
Bad choice of words. I meant queue as "put after".. so I tried to put `refresh_grid` after `GTK_WIDGET_CLASS..->size_allocate (..)` it did flicker a lot (not always) on panel load. I just corrected that, the check to be sure size_allocate is present on parent class, seems to solve it. I guess there's some order of execution on panel load when sometimes the grid size_allocate happens too early. I think the code works really well now. Thank you for your reviews. :)
I'm done for today, unfortunately I can't spend every free time on that everyday.

Reply:
Problem might be that changing / updating grid in size allocate queues new resize...

`if (GTK_WIDGET_CLASS (na_grid_parent_class)->size_allocate)` should not be needed, I think it should never be `NULL`!